common/src/main/java/com/wynntils/features/user/ChatTimestampFeature.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user;
+
+import com.wynntils.core.config.Config;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.StartDisabled;
+import com.wynntils.wynn.event.ChatMessageReceivedEvent;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@StartDisabled
+public class ChatTimestampFeature extends UserFeature {
+    @Config
+    public String formatPattern = "HH:mm:ss";
+
+    @SubscribeEvent
+    public void onChat(ChatMessageReceivedEvent event) {
+        Component message = event.getMessage();
+
+        // use the user specified formatter and if that formatter fails for whatever reason handle it as gracefully as
+        // we can
+        MutableComponent timestamp;
+        try {
+            LocalDateTime date = LocalDateTime.now();
+            DateTimeFormatter formatter = DateTimeFormatter.ofPattern(formatPattern);
+
+            timestamp = new TextComponent("§8[§7" + date.format(formatter) + "§8]§r ");
+        } catch (Exception e) {
+            timestamp = new TextComponent("§8[§7<Invalid time format string>§8]§r ");
```

Comment:
I don't like how we handle this. IMO, check correctness on config update, print the incorrect message once, and have a boolean to return early onChat, if the format is incorrect (or set the cached DateTimeFormatter to null and check for that)

Reply:
would it be better to have the feature soft disable on an invalid formatstring or would it be better to have it revert to the default value with a message telling the user that the value has been reverted?

Reply:
Soft disable cause reverting would be buggy with the config system. Not the greatest solution but oh well.